test/test_polygon.py

```diff
@@ -91,6 +91,43 @@ def test_construction_frompolygon(self):
         self.assertEqual(po_2.vertices, [p1, p2, p3, p4])
         self.assertEqual(po_2.vertices, po.vertices)
 
+    def test_subscript(self):
+        """Checks whether reassigning a vertex works correctly"""
+        po = Polygon([p1, p2, p3, p4])
+
+        po[0] = (70.0, 80.0)
+        self.assertEqual(po[0], (70.0, 80.0))
+
+        po[1] = [45.0, 38.0]
+        self.assertEqual(po[1], (45.0, 38.0))
+
+    def test_length(self):
+        po = Polygon([p1, p2, p3, p4])
+        self.assertEqual(len(po), 4)
+
+        po = Polygon([p1, p2, p3, p4, [75.0, 83.0], [23.0, 12.0], [90.0, 134.0]])
+        self.assertEqual(len(po), 7)
```

Comment:
We have many ways to construct a polygon, might be wise to test all of them to check the size is the same always.